### PR TITLE
examples/filesystem: add support for fatfs on SD card

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This adds support for fatfs to `examples/filesystem`.

This whole vfs/filesystem thing is in dire need of a cleanup.
But add this first to have a test-case that actually works for most supported file systems.

### Testing procedure

Attach an SD card using SPI. The easiest way is to solder jumper wires to one of those ubiquitous microSD -> SD Adapters. You will also have to provide a `sdcard_spi_params.h` for your board - just fill in the SPI pins.

Vss = GND

Now enable `mtd_sdcard` and `fatfs_vfs` - you should be able to `mount` and view the files on the SD card.

Note: VFS uses 32bit for size, so accessing areas beyond 4 GiB is probably broken.

### Issues/PRs references

The example still defaults to littlefs, so 21f8adb0b48bded1a71f63f5738e3b9d8a1deb41 from #14006 is needed to prevent a build failure on boards that have an SD card pre-configured.

### CODE REVIEW

**Comment by bergzand** [comment]
@benpicco Is it possible for you to split 902d6c9ac367aeaceecc29abde0ac29190f925f2 into two commits, one for the SD-card changes and one for FATfs changes?

**Comment by benpicco** [comment]
Sure!

File: examples/filesystem/main.c [hunk 1]
```diff
@@ -26,6 +26,21 @@
...
+
+#define SDCARD_SPI_NUM ARRAY_SIZE(sdcard_spi_params)
+extern sdcard_spi_t sdcard_spi_devs[SDCARD_SPI_NUM];
+mtd_sdcard_t mtd_sdcard_devs[SDCARD_SPI_NUM];
+
+/* always default to first sdcard*/
+static mtd_dev_t *mtd0 = (mtd_dev_t*)&mtd_sdcard_devs[0];
+
+#define MTD_0 mtd0
+#endif
```
Comment (bergzand) [inline examples/filesystem/main.c:42]: Could you add some comments here, similar to the other blocks to explain why every bit is required for SD card operation. It makes the example a bit more useful.

**Comment by bergzand** [comment]
While this PR is probably not to blame, the example hangs indefinitely when no SD-card is present on startup. Inserting an SD-card doesn't allow the firmware to continue, a reset is required.

**Comment by benpicco** [comment]
> While this PR is probably not to blame, the example hangs indefinitely when no SD-card is present on startup. Inserting an SD-card doesn't allow the firmware to continue, a reset is required.

that is a known issue… #11388

File: examples/filesystem/main.c (hunk 1 above)
Comment (benpicco) [reply examples/filesystem/main.c]: I also brought it more in line with other MTD examples.

**Comment by bergzand** [comment]
Looks good, please squash!

**Pull request merged**